br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -196,11 +196,16 @@ func (r *byteReader) switchToConcurrentReader() error {
 // The returned slice (pointer) can not be used after r.reset. In the same interval of r.reset,
 // byteReader guarantees that the returned slice (pointer) will point to the same content
 // though the slice may be changed.
-func (r *byteReader) readNBytes(n int) (*[]byte, error) {
+func (r *byteReader) readNBytes(n int) (*slice, error) {
+	originOffset := r.curBufOffset
 	b := r.next(n)
 	readLen := len(b)
 	if readLen == n {
-		ret := &b
+		ret := &slice{
```

Comment:
maybe return `r.curBuf[originOffset:readLen]`? I will test it in my local environment

Reply:
I'm not sure about the improvements of this PR. Seems there must be something that escapes to heap

```
go build  -gcflags="-m=2"
...
./byte_reader.go:204:10: &slice{...} escapes to heap:
./byte_reader.go:204:10:   flow: ret = &{storage for &slice{...}}:
./byte_reader.go:204:10:     from &slice{...} (spill) at ./byte_reader.go:204:10
./byte_reader.go:204:10:     from ret := &slice{...} (assign) at ./byte_reader.go:204:7
./byte_reader.go:204:10:   flow: {heap} = ret:
./byte_reader.go:204:10:     from append(r.retPointers, ret) (call parameter) at ./byte_reader.go:209:25
```